qa/tasks/cephfs/test_volumes.py

```diff
@@ -4345,6 +4354,407 @@ def test_subvolume_user_metadata_list_and_remove_for_legacy_subvolume(self):
+        # tearDown() so that there's zero chance that it interferes with next
+        # test.
+        self.config_set('mgr', self.conf_opt, False)
+
+        # ensure purge threads have no jobs left from previous test so that
+        # next test doesn't have to face unnecessary complications.
+        self._wait_for_trash_empty()
+
+        super().tearDown()
+
+    def _get_sv_path(self, v, sv):
+        sv_path = self.get_ceph_cmd_stdout(f'fs subvolume getpath {v} {sv}')
+        sv_path = sv_path.strip()
+        # delete slash at the beginning of path
+        sv_path = sv_path[1:]
+
+        sv_path = os.path.join(self.mount_a.mountpoint, sv_path)
+        return sv_path
+
+    def _assert_sv_is_absent_in_trash(self, sv, sv_path, sv_files):
+        uuid = self.mount_a.get_shell_stdout('sudo ls volumes/_deleting').\
+            strip()
+
+        trash_sv_path = sv_path.replace('_nogroup', f'_deleting/{uuid}')
+        trash_sv_path = trash_sv_path.replace(sv, '')
+
+        try:
+            sv_files_new = self.mount_a.get_shell_stdout(
+                f'sudo ls {trash_sv_path}')
+        except CommandFailedError as cfe:
+            # in case dir for subvol including files in it are deleted
+            self.assertEqual(cfe.exitstatus, 2)
+            return
+
+        # in case dir for subvol is undeleted yet (but files inside it are).
+        for filename in sv_files:
+            self.assertNotIn(filename, sv_files_new)
+
+    def test_when_false_subvol_is_trashed_and_also_purged(self):
+        '''
+        Test that when MGR config option mgr/volumes/pause_purging is set to
+        false, running "ceph fs subvolume rm" will move the subvolume to trash
+        & also asynchronously purge it, that is delete it from trash.
+        '''
+        v = self.volname
+        sv = 'sv1'
+
+        self.run_ceph_cmd(f'fs subvolume create {v} {sv} --mode=777')
+        self.config_set('mgr', self.conf_opt, False)
+        sv_path = self._get_sv_path(v, sv)
+        self._do_subvolume_io(sv, number_of_files=500)
+        sv_files = self.mount_a.get_shell_stdout(f'sudo ls {sv_path}').\
+            strip().split('\n')
+
+        self.run_ceph_cmd(f'fs subvolume rm {v} {sv}')
+        time.sleep(5)
+        self._assert_sv_is_absent_in_trash(sv, sv_path, sv_files)
+
+    def _assert_trashed_sv_is_unpurged(self, sv, sv_path, sv_files):
+        uuid = self.mount_a.get_shell_stdout('sudo ls volumes/_deleting').\
+                    strip()
+
+        trash_sv_path = sv_path.replace('_nogroup', f'_deleting/{uuid}')
+        trash_sv_path = trash_sv_path.replace(sv, '')
+        sv_files_new = self.mount_a.get_shell_stdout(f'sudo ls {trash_sv_path}').\
+            strip()
+
+        if len(sv_files) == len(sv_files_new):
+            return
+        else:
+            for filename in sv_files:
+                self.assertIn(filename, sv_files_new)
+
+    def test_when_true_subvol_is_trashed_but_stays_unpurged(self):
```

Comment:
Similarly, we can have like:`test_subvolume_when_pause_purging_trash_is_enabled`. The goal of the test case is anyways mentioned in the description

Reply:
In logs test names appears like this: `tasks.cephfs.test_volumes.TestPausePurging.test_when_true_subvol_is_trashed_but_stays_unpurged`, so adding config option name is test name again is not needed because test class already mentioned it.